### PR TITLE
Update the build library tag version for build job

### PR DESCRIPTION
**Pull request by zelinh**
### Description
Update the build library tag version for build job

### Issues Resolved
https://github.com/opensearch-project/opensearch-build-libraries/issues/392

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by gaiksaya** [comment]
Hi @zelinh Can you please rebase and push? Need to see if linkchecker is passing now

**Comment by zelinh** [comment]
> Hi @zelinh Can you please rebase and push? Need to see if linkchecker is passing now

Seems like after rebase, it's still failing.

**Comment by gaiksaya** [comment]
> > Hi @zelinh Can you please rebase and push? Need to see if linkchecker is passing now
> 
> Seems like after rebase, it's still failing.

Looks like it is still checking release notes folder. 


Wondering if it a syntax issue?
```
--exclude-path "tests/" "./" "release-notes/"
```

**Comment by zelinh** [comment]
> > > Hi @zelinh Can you please rebase and push? Need to see if linkchecker is passing now
> > 
> > 
> > Seems like after rebase, it's still failing.
> 
> Looks like it is still checking release notes folder.
> 
> Wondering if it a syntax issue?
> 
> ```
> --exclude-path "tests/" "./" "release-notes/"
> ```

Should it be like `--exclude-path "tests/"  --exclude-path "release-notes/" "./"`

Or what we could do is create a new `. lycheeignore ` file according to their suggestion. https://github.com/lycheeverse/lychee-action#excluding-links-from-getting-checked

**Comment by gaiksaya** [comment]
Looks like your solution worked @zelinh 

Can you please fix the actual errors?

```
### Errors in ./docs/Testing-the-Distribution.md

* [404] [https://github.com/opensearch-project/opensearch-build/blob/main/manifests/2.11.1/opensearch-dashboards-2.11.1-test.yml](https://github.com/opensearch-project/opensearch-build/blob/main/manifests/2.11.1/opensearch-dashboards-2.11.1-test.yml) | Failed: Network error: Not Found
* [404] [https://github.com/opensearch-project/opensearch-build/blob/main/manifests/2.11.1/opensearch-2.11.1-test.yml](https://github.com/opensearch-project/opensearch-build/blob/main/manifests/2.11.1/opensearch-2.11.1-test.yml) | Failed: Network error: Not Found

### Errors in ./docs/Releasing-the-Distribution.md

* [404] [https://github.com/opensearch-project/opensearch-build/blob/main/manifests/2.9.0/opensearch-2.9.0.yml](https://github.com/opensearch-project/opensearch-build/blob/main/manifests/2.9.0/opensearch-2.9.0.yml) | Failed: Network error: Not Found
* [404] [https://github.com/issues?q=is%3Aopen+is%3Aissue+user%3Aopensearch-project+%5BRELEASE%5D+Release+version+2.10.0+in%3Atitle+](https://github.com/issues?q=is%3Aopen+is%3Aissue+user%3Aopensearch-project+%5BRELEASE%5D+Release+version+2.10.0+in%3Atitle+) | Failed: Network error: Not Found
* ```

**Comment by zelinh** [comment]
> Looks like your solution worked @zelinh
> 
> Can you please fix the actual errors?
> 
> ```
> ### Errors in ./docs/Testing-the-Distribution.md
> 
> * [404] [https://github.com/opensearch-project/opensearch-build/blob/main/manifests/2.11.1/opensearch-dashboards-2.11.1-test.yml](https://github.com/opensearch-project/opensearch-build/blob/main/manifests/2.11.1/opensearch-dashboards-2.11.1-test.yml) | Failed: Network error: Not Found
> * [404] [https://github.com/opensearch-project/opensearch-build/blob/main/manifests/2.11.1/opensearch-2.11.1-test.yml](https://github.com/opensearch-project/opensearch-build/blob/main/manifests/2.11.1/opensearch-2.11.1-test.yml) | Failed: Network error: Not Found
> 
> ### Errors in ./docs/Releasing-the-Distribution.md
> 
> * [404] [https://github.com/opensearch-project/opensearch-build/blob/main/manifests/2.9.0/opensearch-2.9.0.yml](https://github.com/opensearch-project/opensearch-build/blob/main/manifests/2.9.0/opensearch-2.9.0.yml) | Failed: Network error: Not Found
> * [404] [https://github.com/issues?q=is%3Aopen+is%3Aissue+user%3Aopensearch-project+%5BRELEASE%5D+Release+version+2.10.0+in%3Atitle+](https://github.com/issues?q=is%3Aopen+is%3Aissue+user%3Aopensearch-project+%5BRELEASE%5D+Release+version+2.10.0+in%3Atitle+) | Failed: Network error: Not Found
> * 
> ```
> 
> The manifests are moved to legacy folder thats why it threw error

This links is accessible. Not sure why it failed. Fixed the rest of those.
https://github.com/issues?q=is%3Aopen+is%3Aissue+user%3Aopensearch-project+%5BRELEASE%5D+Release+version+2.10.0+in%3Atitle+

**Review by gaiksaya — APPROVED** [review]

File: .github/workflows/link-checker.yml [hunk 1]
```diff
@@ -14,7 +14,7 @@ jobs:
         id: lychee
         uses: lycheeverse/lychee-action@v1
         with:
-          args: --verbose --max-retries 5 --retry-wait-time 10 --accept=200,403,429 --exclude-path tests/ ./ release-notes/
+          args: --verbose --max-retries 5 --retry-wait-time 10 --accept=200,403,429 --exclude-path tests/ --exclude-path release-notes/ ./
```
Comment (peterzhuamazon) [inline .github/workflows/link-checker.yml:17]: Incorrect rebase?

Comment (zelinh) [reply .github/workflows/link-checker.yml:17]: Seems like the syntax is not correct. Trying to fix it.

File: docs/Releasing-the-Distribution.md [hunk 2]
```diff
@@ -123,7 +123,7 @@ This section is not required for a patch release.
...
+The component release issues are auto created by the workflows part of the build repo [OpenSearch components](https://github.com/opensearch-project/opensearch-build/blob/main/.github/workflows/os-release-issues.yml), [OpenSearch Dashboards components](https://github.com/opensearch-project/opensearch-build/blob/main/.github/workflows/osd-release-issues.yml). These workflows create the release issues based on the template [component_release_template.md](https://github.com/opensearch-project/opensearch-build/blob/main/.github/ISSUE_TEMPLATE/component_release_template.md) and links back the global release issue part of the build. Sample component release issues created for [2.10.0] Sample link `https://github.com/issues?q=is%3Aopen+is%3Aissue+user%3Aopensearch-project+%5BRELEASE%5D+Release+version+2.10.0+in%3Atitle+`.
```
Comment (zelinh) [inline docs/Releasing-the-Distribution.md]: Seems like this link is not accessible to non-login user. Mark it as code.

**Pull request merged**